src/relay.ts

```diff
@@ -38,11 +38,24 @@ async function getWalletClient(): Promise<WalletClient> {
 export default async function relay(
   relayerAddress: string,
   rateFeedName: string,
+  network: string,
 ): Promise<boolean> {
-  const logger = getLogger(rateFeedName);
+  const logger = getLogger(rateFeedName, network);
   const publicClient = getPublicClient();
   const wallet = await getWalletClient();
 
+  // Check if the address is a contract
+  const contractCode = await publicClient.getCode({
+    address: relayerAddress as Address,
+  });
+
+  if (!contractCode || contractCode === "0x") {
+    logger.error(
+      `Relay failed. Relayer address ${relayerAddress} is not a contract.`,
+    );
+    return false; // Not a contract
+  }
```

Comment:
Given that this should only happen because of a misconfiguration of the relayer address, what do you think of converting this into an assert statement instead?

Reply:
hmm, what would be the advantage of that?